tests/simple/simple_test.go

```diff
@@ -113,30 +130,95 @@ func TestMain(m *testing.M) {
-				desc := fmt.Sprintf("%s/%s/%s", testFile.Name, section.Name, test.Name)
-				t.Run(desc, func(t *testing.T) {
-					err := rc.RunTest(test)
-					if err != nil {
-						t.Fatal(err)
+		allSections := strings.Split(flagVal, ";")
+		filename := flagVal[:fileInd+1]
+		for _, sectionVal := range allSections {
+			if firstFlag {
+				skipTests = append(skipTests, sectionVal)
+				firstFlag = false
+			} else {
+				sections := strings.Count(sectionVal, "/")
+				if sections == 0 {
+					if sectionVal != "" {
+						skipTests = append(skipTests, filename + sectionVal)
+					} else {
+						log.Fatal("Empty string where should be section name")
 					}
-				})
+				} else if sections == 1 {
+					lastInd := strings.LastIndex(sectionVal, "/")
+					sectionName := sectionVal[:lastInd+1]
+					testString := sectionVal[lastInd+1:]
+					if testString == "" {
+						log.Fatal("Empty string where should be test name")
+					} else {
+						tests := strings.Split(testString, ",")
+						for _, test := range tests{
+							if test == "" {
+								log.Fatal("Empty string where should be test name")
+							} else {
+								skipTests = append(skipTests, filename + sectionName + test)
+							}
+						}
+					}
+				} else {
+					log.Fatal("Unable to parse skip_test flag, specifically for ", sectionVal)
+				}
 			}
 		}
+		firstFlag = true
 	}
+        // Run the flag-configured tests.
+        for _, filename := range flag.Args() {
+                testFile, err := parseSimpleFile(filename)
+                if err != nil {
+                        t.Fatalf("Can't parse input file %v: %v", filename, err)
+                }
+                t.Logf("Running tests in file %v\n", testFile.Name)
+                for _, section := range testFile.Section {
+			sectionPath := testFile.Name + "/" + section.Name
+                        if contains(skipTests, sectionPath) {
+                                t.Logf("Skipping all tests in section %v\n", section.Name)
+                        } else {
+                                t.Logf("Running tests in section %v\n", section.Name)
+                                for _, test := range section.Test {
+					testPath := sectionPath + "/" + test.Name
+                                        if contains(skipTests, testPath){
+						t.Logf("Skipping test name %v\n", test.Name)
+					} else {
+                                                desc := fmt.Sprintf("%s/%s/%s", testFile.Name, section.Name, test.Name)
+                                                t.Run(desc, func(t *testing.T) {
+                                                        err := rc.RunTest(test)
+                                                             if err != nil {
+                                                                t.Fatal(err)
+                                                        }
+						})
+                                        }
+				}
+			}
+		}
+	}
+}
+func contains(s []string, e string) bool {
```

Comment:
Double-check if there's a utility somewhere that already does this.  Add a comment that we could store the strings in a sorted order to speed up the checks if needed.